samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

```diff
@@ -0,0 +1,351 @@
+This sample also demonstrates how to implement error tolerance in your cellular applications without relying on reboot loops.
+
+.. _nrf_cloud_mqtt_multi_services_sample_requirements:
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :sample-yaml-rows:
+
+.. _nrf_cloud_mqtt_multi_services_sample_features:
+
+Features
+********
+
+This sample implements or demonstrates the following features:
+
+* Error-tolerant use of the `nRF Cloud MQTT API`_ using the :ref:`nrf_modem <nrfxlib:nrf_modem>` and :ref:`nrf_cloud <lib_nrf_cloud>` libraries.
+* Support for `Firmware Over The Air (FOTA) <nRF Cloud Getting Started FOTA documentation_>`_ via the `nRF Cloud`_ portal.
+* Support for `modem AT Commands <AT Commands Reference Guide_>`_ over UART using the :ref:`AT Host library<lib_at_host>`.
+* Periodic cellular and GNSS location tracking using the :ref:`location library <lib_location>`.
+* Periodic temperature sensor sampling on your `Nordic Thingy:91`_, or fake temperature  measurements on your `Nordic nRF9160 DK`_.
+* Transmission of sensor and GNSS location samples to the `nRF Cloud`_ portal as `nRF Cloud device messages <nRF Cloud Device Messages_>`_.
+* Construction of valid `nRF Cloud device messages <nRF Cloud Device Messages_>`_ using `cJSON`_.
+
+
+.. _nrf_cloud_mqtt_multi_services_sample_structure:
+.. _nrf_cloud_mqtt_multi_services_sample_structure_and_theory:
+.. _nrf_cloud_mqtt_multi_services_sample_structure_and_theory_of_operation:
+.. _nrf_cloud_mqtt_multi_services_sample_theory_of_operation:
+
+Structure and Theory of Operation
+*********************************
+
+This sample is separated into a number of smaller functional units.
+The top level functional unit and entry point for the sample is :file:`src/main.c`.
+This file starts three threads, each with a distinct function:
+
+* Connection Thread (``conn_thread``): Runs the `Connection Management Loop <nrf_cloud_mqtt_multi_services_sample_connection_management_loop_>`_, which maintains a connection to `nRF Cloud`_.
+* Application Thread (``app_thread``): Runs the `Main Application Loop <nrf_cloud_mqtt_multi_services_sample_main_application_loop_>`_, which may add `device messages <nRF Cloud Device Messages_>`_ to the `Device Message Queue <nrf_cloud_mqtt_multi_services_sample_device_message_queue_>`_.
+* Message Thread (``msg_thread``): Processes the `Device Message Queue <nrf_cloud_mqtt_multi_services_sample_device_message_queue_>`_ whenever there is an active connection.
+
+.. _nrf_cloud_mqtt_multi_services_sample_connection_management_loop:
+
+Connection Management Loop
+==========================
+
+The Connection Management Loop performs four primary tasks:
+
+1) Configure and activate the cellular modem.
+#) Start a connection to the LTE Network.
+#) Maintain a connection to `nRF Cloud`_.
+#) Reestablish or retry all lost or failed connections.
```

Comment:
Believe it or not, this is actually a sequential set of tasks, -- they must be performed in this order, and this is the order in which the connection loop performs them. 

It's a subtle suggestion to the reader, for sure, using a numbered list, but someone familiar with networking concepts may pick up on it and find it useful. I don't explicitly state that the items are in order, because I worry that the 4th item would create some confusion for someone less familiar with networking concepts (because it, technically, happens at all stages throughout the loop in various capacities).

Does this change your mind on the matter, or do you still think it would be clearer to have this be an unordered list?

Reply:
Fine to have the list numbered if the things take place in that order (or rather the connection loop performs these tasks). But change the verb to third person form.